nomad/state/state_store_test.go

```diff
@@ -1661,6 +1661,70 @@ func TestStateStore_DeleteJob_Job(t *testing.T) {
 	}
 }
 
+func TestStateStore_DeleteJobTxn_BatchDeletes(t *testing.T) {
+	state := testStateStore(t)
+
+	const testJobCount = 10
+	const jobVersionCount = 4
+
+	jobs := make([]*structs.Job, testJobCount)
+	for i := 0; i < testJobCount; i++ {
+		job := mock.BatchJob()
+
+		err := state.UpsertJob(uint64(100+i), job)
+		require.NoError(t, err)
+
+		jobs[i] = job
+
+		// create some versions
+		for vi := 1; vi < jobVersionCount; vi++ {
+			job := job.Copy()
+			job.TaskGroups[0].Tasks[0].Env = map[string]string{
+				"Version": fmt.Sprintf("%d", vi),
+			}
+
+			require.NoError(t, state.UpsertJob(uint64(100+i+vi*10), job))
```

Comment:
nit - the upsert index logic seems unnecessarily detailed (100+i+vi*10), I would start a variable with a value like 1000 and then add one each time in the loop.